helix-term/src/application.rs

```diff
@@ -419,21 +421,27 @@ impl Application {
         }
     }
 
-    pub fn handle_terminal_events(&mut self, event: Result<Event, crossterm::ErrorKind>) {
+    pub fn handle_terminal_events(&mut self, event: Result<Input, crossterm::ErrorKind>) {
         let mut cx = crate::compositor::Context {
             editor: &mut self.editor,
             jobs: &mut self.jobs,
             scroll: None,
         };
         // Handle key events
         let should_redraw = match event {
-            Ok(Event::Resize(width, height)) => {
-                self.compositor.resize(width, height);
-
-                self.compositor
-                    .handle_event(Event::Resize(width, height), &mut cx)
-            }
-            Ok(event) => self.compositor.handle_event(event, &mut cx),
+            Ok(input) => match input {
+                Input::Event(Event::Resize(width, height)) => {
+                    self.compositor.resize(width, height);
+                    self.compositor
+                        .handle_event(Event::Resize(width, height), &mut cx)
+                }
+                Input::Event(event) => self.compositor.handle_event(event, &mut cx),
+                Input::Paste(data) => {
+                    let (view, doc) = current!(self.editor);
+                    crate::commands::paste_bracketed_value(data, doc, view);
```

Comment:
Not sure where best to stick this call. Doesn't seem like something that should be on compositor?